### DISCUSSION TITLE
How to serve `/admin` from different URL for obscurity?

### DISCUSSION THREAD

**Discussion by jjaimealeman** [directus/directus#12507] [Archive]
I came across this discussion from one year ago - but I feel it was/is unresolved?!

`Add way to configure redirect to /admin #4607` https://github.com/directus/directus/discussions/4607

According to the docs, it should be possible: https://docs.directus.io/configuration/config-options/#general
![screenshot](https://i.imgur.com/CcN4mxd.png)

Has anyone come up with a simple way to serve `/admin` from another URL?

**AND**

Prevent root_url from redirecting to `/admin`?

**Comment by scolastico** [comment]
I think you could run (like in the dev environment) the api and app separately. Take a look here: https://docs.directus.io/contributing/running-locally/#_7-start-the-development-server

**Reply by jjaimealeman** [reply]
Thanks for the reply.
Unfortunately, this Directus instance is running on a live server hosted on DigitalOcean.

I have two domains:
`www.example.com` Front-end w/Nuxt3 on Netlify.
`www.examples.com` Back-end Directus on DigitalOcean.

I have several options regarding safe-guarding the Back-end:
1. a sub-domain for Directus
2. renaming `/admin`
3. disabling the redirect to `/admin`

Sadly I don't know how to accomplish either one - as I'm just wanting to make it more secure ;)

**Reply by scolastico** [reply]
I see there 2 options you could configure a reverse proxy with for example Apache to block requests to `/admin` and create for a other path an additional reverse proxy to `/admin`. Or you could split the app and API apart like i previously suggested and reverse proxy for example `my-great-admin-panel.example.com` to the app and `my-great-cms.example.com` to the API. If you configure anything correct both solutions should work. Also trust me renaming the admin URL doesn't gives you that much of security the better approach would be to use for example reCaptcha for blocking brute force and using good passwords.

**Reply by scolastico** [reply]
Also if you don't have a clue how to configure reverse proxys check this out: https://nginxproxymanager.com

**Reply by MarvinXu** [reply]
This way the image inserted in WYSIWYG editor is prefixed `my-great-admin-panel.example.com` which will be exposed in frontend

**Comment by azrikahar** [comment]
> Has anyone come up with a simple way to serve /admin from another URL?

There isn't an out of the box config for this yet, but there was a similar question and my answer here does revolve around the reverse proxy approach mentioned above: https://github.com/directus/directus/discussions/10159#discussioncomment-2033583. 
 
As for making it configurable, there is the draft PR #10641. It's just not being actively worked on at the moment 👍

> Prevent root_url from redirecting to /admin?

Does setting the `ROOT_REDIRECT` to `false` not prevent it? I believe it should. 🤔

**Reply by rijkvanzanten** [reply]
> Does setting the ROOT_REDIRECT to false not prevent it? I believe it should. 🤔

It prevents the root path `/` from redirecting to `/admin`, but it doesn't allow you to change the name of `/admin` 👍🏻

**Reply by azrikahar** [reply]
Ya I was inquiring about the redirecting portion, not the name change. 😄 OP's description seemed like saying the redirecting is not working.

**Reply by jjaimealeman** [reply]
> OP's description seemed like saying the redirecting is not working.

My mistake. Poorly worded on my part I guess.

The `/admin` redirect is working fine, I was just wanting to find a way to change `/admin` to `/more-secure-url` 😀 

It seems the solution you posted #10159 may be my only choice. I'm just not too familiar with Nginx/Caddy/Docker (_yet!_)

**Comment by vogdb** [comment]
Hi! Would it be possible to replace `/admin` with another `/subfolder` using Nginx only? Or it is impossible without adjusting Directus code?
